### PR TITLE
feature: support multi service discovery

### PR DESCRIPTION
**Pull request by shoogoome**
### What this PR does / why we need it:
support multi service discovery

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Comment by membphis** [comment]
@qiujiayu do you have time to look at this PR?

**Comment by membphis** [comment]
@shoogoome please take a look at the output of CI: https://github.com/apache/apisix/pull/2556/checks?check_run_id=1324835665#step:10:187

**Comment by ShiningRush** [comment]
You should init dicovery at https://github.com/apache/apisix/blob/master/apisix/init.lua#L94.

File: t/discovery/eureka.t [hunk 1]
```diff
@@ -26,7 +26,8 @@ apisix:
   node_listen: 1984
   config_center: yaml
   enable_admin: false
-  discovery: eureka
+  discovery:
+    - eureka
```
Comment (ShiningRush) [inline t/discovery/eureka.t:31]: We can also move all eureka configuration to here, it would be better.

**Review by ShiningRush — CHANGES_REQUESTED** [review]

**Comment by shoogoome** [comment]
> @shoogoome please take a look at the output of CI: https://github.com/apache/apisix/pull/2556/checks?check_run_id=1324835665#step:10:187

thx

File: t/discovery/eureka.t (hunk 1 above)
Comment (shoogoome) [reply t/discovery/eureka.t]: done

File: apisix/discovery/eureka.lua [hunk 2]
```diff
@@ -66,7 +66,8 @@ local _M = {
 
 
 local function service_info()
-    local host = local_conf.eureka and local_conf.eureka.host
+    local host = local_conf.discovery and
+        local_conf.discovery.eureka and local_conf.discovery.eureka.host
```
Comment (nic-chen) [inline apisix/discovery/eureka.lua:70]: Could we support different instances for a same type discovery?

Comment (ShiningRush) [reply apisix/discovery/eureka.lua:70]: This scenario is unreasonable and will not be supported for now. 
However, users can still handle this scenario by creating new instances of different gateways, I think it is enough.

**Review by ShiningRush — APPROVED** [review]

**Review by membphis — APPROVED** [review]
LGTM

**Review by ShiningRush — CHANGES_REQUESTED** [review]
Oh, we also need to modify document of dicovery

**Review by moonming — APPROVED** [review]

File: doc/discovery.md [hunk 3]
```diff
@@ -167,31 +168,33 @@ The supported discovery client: Eureka.
...
-    connect: 2000                  # 2000ms
-    send: 2000                     # 2000ms
-    read: 5000                     # 5000ms
+discovery:
+  eureka:
+    host:                            # it's possible to define multiple eureka hosts addresses of the same eureka cluster.
+      - "http://${usename}:${passowrd}@${eureka_host1}:${eureka_port1}"
+      - "http://${usename}:${passowrd}@${eureka_host2}:${eureka_port2}"
+    prefix: "/eureka/"
+    fetch_interval: 30               # 从 eureka 中拉取数据的时间间隔，默认30秒
```
Comment (ShiningRush) [inline doc/discovery.md:177]: we should using English to describe it.

**Review by ShiningRush — CHANGES_REQUESTED** [review]

**Review by ShiningRush — APPROVED** [review]

**Pull request merged**